zanata-war/src/main/java/org/zanata/action/LoginAction.java

```diff
@@ -105,9 +116,35 @@ public String login() {
                     "login() only supports internal, jaas, or kerberos authentication");
         }
 
+        if ("loggedIn".equals(loginResult)) {
+            if (authenticationManager.isAuthenticated() && authenticationManager.isNewUser()) {
+                return "createUser";
+            }
+            if (authenticationManager.isAuthenticated() && !authenticationManager.isNewUser() && userRedirect.shouldRedirectToDashboard()) {
+                return "dashboard";
+            }
+            if (authenticationManager.isAuthenticated() && !authenticationManager.isNewUser() && userRedirect.isRedirect()) {
+                // TODO [CDI] seam will create a conversation when you return view id directly or redirect to external url
+                return continueToPreviousUrl();
+            }
+        } else if ("inactive".equals(loginResult)) {
+            Conversation.instance().end();
+            return "inactive";
+        }
```

Comment:
Yes. From memory faces-config has less functionality than pages.xml. e.g. it seems only work with string result not just from different EL.